### ISSUE TITLE
General: Fix accessibility issues on Green color usage

### ISSUE THREAD

**Issue by nuriapenya** [Automattic/wp-calypso#101301]
## Problem

We spotted few accessibility issues when using the green colour. 

1. We've recently introduced a new badge style on the Plans Grid that is inaccessible 
2. Text within domains page page
3. Text within domain section in /overview
4. The green discount on dark mode (e.g. Hosting plans grid).

<img width="731" alt="Image" src="https://github.com/user-attachments/assets/cb0cd8e7-4ae5-405a-9609-3a9127b32095" />

<img width="2053" alt="Image" src="https://github.com/user-attachments/assets/fe7dced0-033b-4154-8f87-f6d002acc9d2" />

<img width="1754" alt="Image" src="https://github.com/user-attachments/assets/5fb5a827-96d8-4821-91f7-658433d37114" />

<img width="1575" alt="Image" src="https://github.com/user-attachments/assets/af9d845a-0d38-423f-ac87-3676aeb6ed08" />

## Solutions 

Specific to the **Pricing grid badge**: It should use the same styles we have in the other green badges:

```
color: var(--studio-green-80);
background-color: var(--studio-green-5);
```

For all the instances when the **text appears alone**, always use:

```
color: var(--studio-green-60);
```

For all the instances when we have **green on dark mode**, always use:

```
color: var(--studio-green-10);
```

> [!NOTE]  
> In a separate issue we will be working on aligning all the badges to Core's [component](https://wordpress.github.io/gutenberg/?path=/docs/components-badge--docs#success)

**Comment by mirka** [comment]
@nuriapenya There seem to be many more instances in the repo where `green-50` is used as the text (or icon) color on a light/dark background. Is it worth doing an overall sweep of these usages, or are we just focusing on the "First-Time Experience" screens listed in this issue?

**Comment by nuriapenya** [comment]
@mirka Ideally we fix everything at once. 🙏🏼 When in doubt, feel free to share some of those instances in this issue and we can look wether there's a good reason son keep `green-50`, but in general, I'd say we could do an overall sweep.

**Comment by mirka** [comment]
Noting another WCAG-failing green badge pattern using Jetpack color tokens.

<img width="500" alt="Boost modal in A4A" src="https://github.com/user-attachments/assets/52712693-5465-4b7c-bb2a-47f9d382766a" />

<img width="300" alt="Screenshot demonstrating failing color contrast" src="https://github.com/user-attachments/assets/d897ed50-b2cc-4c38-a0aa-07f1f86e19fe" />

(Go to `https://agencies.automattic.com/sites` and click the Add Boost button on one of your sites to reproduce.)

https://github.com/Automattic/wp-calypso/blob/a36ee66e94a4aa1fe0eeb67a07ac1f33457c53db/client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content/site-boost-column/style.scss#L42-L43

I understand this is an upsell modal for a Jetpack feature, but not sure if it's necessary to use Jetpack brand colors for these badges in the first place.

**Comment by mirka** [comment]
I'll also flag that the default Jetpack shade is set to `40` (`#069e08`) in Color Studio, which does not have sufficient contrast against `#ffffff`.

https://github.com/Automattic/color-studio/blob/556fdfb55262ab1396b478684a3dc84a954182fe/data/color-definitions.js#L190-L191

I guess we'll want to deal with Jetpack greens as a separate issue? Might be a different can of worms.

**Comment by crisbusquets** [comment]
@keoshi 👆 (Jetpack badges).

@mirka I think we should tackle Jetpack ones as a separate issue, yes.

**Comment by keoshi** [comment]
> but not sure if it's necessary to use Jetpack brand colors for these badges in the first place.

Agreed we don't need those to be Jetpack green. Whatever y'all land on for Dotcom I'm sure it would be a good improvement over on Jetpack.com as well.

**Comment by nuriapenya** [comment]
Added two issues to fix:

[Checkout: Update green colour (accessibility fix) #102776](https://github.com/Automattic/wp-calypso/issues/102776)

[Cart: Update green colour (accessibility fix) #102777](https://github.com/Automattic/wp-calypso/issues/102777)

@mirka is this something you or the team could help fix as part of Code Blue? 🙏🏼

**Comment by mirka** [comment]
I'm afraid our team is not able to prioritize these specific fixes right now.

But I do want to mention that the way we've been addressing these contrast issues is not quite ideal. We've simply been swapping out certain shades with other shades, without adding any prevention measures to keep these problems from cropping up again. There is still no way for anyone to reasonably know which shades are supposed to be used when, so these issues will likely reoccur after we fix the existing ones.

A more systemic fix will involve things like improving documentation, having better color tokens (or appropriately named alias tokens) that hint at when they're supposed to be used, and including these "semantic colors" (green, yellow, red) as part of the theming system that @ciampo and @jameskoster are leading right now. These systemic efforts will continue to be worked on by @Automattic/team-calypso.

**Comment by jboland88** [comment]
@Copons I see you are assigned on the main issue and I am on the sub issues :)
Would you prefer to handle all these in one go?
Happy to fix the two specific sub issues individually as well

**Comment by Copons** [comment]
@josh.boland feel free to reassign the sub-issues to me!
